elijah/README.md

```diff
@@ -0,0 +1,305 @@
+    
+    var sponge = function(s) {
+        // take in string and append "spongebob squarepants!"
+        return s = '\n' + spongeBob + '\n';
+    }
+    
+    var songBuilder = function(a) {
+        // build song string from array a
+        for (i = 0; i < a.length; i++) {
+            outputSong += a[i];
+            outputSong += sponge(outputSong);
+        }
+        for (i = 0; i < 2; i++) {
+            outputSong += sponge(outputSong);
+        }
+        return outputSong += '\n' + spongeBob.toUpperCase();
+    }
+
+    // "play" song
+    console.log(songBuilder(song));
+
+---
+
+**4)** _(Difficult, 20%)_
+
+Suppose the '&' key on your keyboard is missing, and you want to be able to continue programming without it.  You decide to simulate the && operator with a function!
+
+**a)**
+Write a function `and2(a,b)` which tries to simulate the && operator: it should always return the same result as `(a && b)` for any values of _a_ and _b_.  (For example, `and2((0>1),true)` should return _false_.)  But you can't use && itself within your function!
+
+
+    var and2 = function(a,b) {
+        return a ? (b ? true : false) : false
+    }
+
+**b)** Write another function `and3(a,b,c)` which tries to simulate a double-&& operator: it should always return the same result as `(a && b && c)` for any values of a,b,c.  (For example, `and3((1>0),(0>1),true)` should return _false_.)  As before, you're not allowed to use && itself. Remember that you can call functions from functions!
+
+
+    var and3 = function(a,b,c) {
+        return and2(a,b) ? (c ? true : false) : false;
+    }
+
+**c)**
+Now generalize your function to handle any number of values.  You will learn better ways eventually, but for now use an array to store all the values.
+Your new function `andN(values)` should accept one parameter _values_, which is an array holding all the values to && together.
+If the argument you provide when calling the function holds values [a,b,c...z], as in
+`andN([a,b,c...z])`, the function should return the same result as the expression `(a && b && c && ... z)`.
+Make sure to handle two special cases: length 0 (then return _true_) and length 1 (then return that single value).
+Again, don't use &&.
+
+    var andN = function(n) {
+        // XXX not optimal, but works... brain dead at moment
+        var a = true;
+        if (n.length === 1) {
+            return n[0];
+        } else {
+            while (a === true) {
+                for (var i = 0; i <= n.length; i++) {
+                    if (i < n.length - 1) {
+                        if (and2(n[i],n[(i + 1)])) {
+                            a = true;
+                        } else {
+                            return a = false;
+                        }
+                    } else {
+                        return a;
+                    }
+                }
+            }
+        }
+    }
+
```

Comment:
This makes sense logically. Here's a simpler solution: 

function andN(values) {
	var n = values.length;
  	for (var i=0; i<n; i++) {
    	if (!values[i]) {
			return values[i]; //return first falsish val
		}
  	}
  	if (n>0) {
    	return values[n-1]; //or else last (truish) val
    }
    return true; // only when n is 0;
}